src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -166,15 +166,7 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
                 throw new ArgumentException("The characters '*' and ',' are not supported in label filters.", nameof(labelFilter));
             }
 
-            if (!_kvSelectors.Any(s => string.Equals(s.KeyFilter, keyFilter) && string.Equals(s.LabelFilter, labelFilter)))
-            {
-                _kvSelectors.Add(new KeyValueSelector
-                {
-                    KeyFilter = keyFilter,
-                    LabelFilter = labelFilter
-                });
-            }
-
+            _kvSelectors.AppendUnique(keyFilter, labelFilter);
             return this;
         }
 
```

Comment:
The problem still exists in `SelectSnapshot` API. We are not preserving precedence order for snapshots.

Reply:
fixed. but next step is considering whether to use a generic utility for it.